diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 7063da0f2..b406f747f 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -432,7 +432,7 @@ default_compiler_configs = [
   "//build/config/compiler:default_symbols",
   "//build/config/compiler:export_dynamic",
   "//build/config/compiler:no_exceptions",
-  "//build/config/compiler:no_rtti",
+  "//build/config/compiler:rtti",
   "//build/config/compiler:runtime_library",
   "//build/config/compiler:thin_archive",
   "//build/config/coverage:default_coverage",
diff --git a/config/android/BUILD.gn b/config/android/BUILD.gn
index 233e8b0e7..1ed3a6bd6 100644
--- a/config/android/BUILD.gn
+++ b/config/android/BUILD.gn
@@ -109,11 +109,16 @@ config("runtime_library") {
                 rebase_path("$android_ndk_root/sources/android/support/include",
                             root_build_dir) ]
 
+  ######### !!!!!Add this to enable build with NDK libc++ !!!!!################################################################
+  cflags_cc += [ "-stdlib=libc++",
+      "-isystem" + rebase_path("$libcxx_prefix/include", root_build_dir),
+      "-isystem" + rebase_path("$libcxxabi_prefix/include", root_build_dir) ]
+
   defines = [
     "__GNU_SOURCE=1",  # Necessary for clone().
     "CHROMIUM_CXX_TWEAK_INLINES",  # Saves binary size.
   ]
-  ldflags = [ "-nostdlib" ]
+  ldflags = ["-Wl,--allow-multiple-definition"]
   lib_dirs = [ android_libcpp_lib_dir ]
 
   libs = []
diff --git a/config/android/config.gni b/config/android/config.gni
index 58838421c..4108a67ab 100644
--- a/config/android/config.gni
+++ b/config/android/config.gni
@@ -46,7 +46,7 @@ if (is_android || is_chromeos) {
     enable_chrome_android_internal = has_chrome_android_internal
 
     # Android API level for 32 bits platforms
-    android32_ndk_api_level = 16
+    android32_ndk_api_level = 21
 
     # Android API level for 64 bits platforms
     android64_ndk_api_level = 21
diff --git a/config/c++/c++.gni b/config/c++/c++.gni
index b639ca779..a41558d00 100644
--- a/config/c++/c++.gni
+++ b/config/c++/c++.gni
@@ -61,5 +61,5 @@ if (is_android && export_libcxxabi_from_executables) {
   libcxx_is_shared = true
 }
 
-libcxx_prefix = "//buildtools/third_party/libc++/trunk"
-libcxxabi_prefix = "//buildtools/third_party/libc++abi/trunk"
+libcxx_prefix = "//third_party/android_ndk/sources/cxx-stl/llvm-libc++"
+libcxxabi_prefix = "//third_party/android_ndk/sources/cxx-stl/llvm-libc++abi"
